### PR TITLE
added prometheus format support

### LINKED ISSUE #26: Support for Play 2.7 etc
Hi together,
showing some love for my initial work (it's been a while...), I started updating the lib for Play 2.7 and added support for the prometheus metrics format. It's not ready for a PR yet, but I am willing to get it there if you would be open to accept a PR? Just wanted to ask before putting too much work in.

All the best,
Lena

### PR DESCRIPTION
**Pull request by lenalebt**
Hi,
I added prometheus format support since I needed it for a toy project and added some tests as well. And I removed some of the toil that was (imho) unneeded, e.g. tut (since no code is there anymore in the README).

I hope it is useful. Closes #26 

All the best,
Lena

### CODE REVIEW

**Comment by musiKk** [comment]
"First-time contributor"?? I think not! 😄

**Comment by lenalebt** [comment]
Yeah, maybe not ;).

**Comment by matterche** [comment]
@lenalebt thanks for the contribution. Will have a look. Btw. do you have a link to a resource which describes the prometheus format?

**Comment by lenalebt** [comment]
Hi @matterche - the format is described here: https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md

Thanks for asking for it, I had a closer look and changed a few details again that were not accurate beforehand. Not everything is 100% mappable, but I think it comes close enough so far.

File: app/org/zalando/markscheider/MetricsController.scala [hunk 1]
```diff
@@ -2,26 +2,70 @@ package org.zalando.markscheider
...
-  def serialize(mapper: ObjectMapper): Result = {
+class MetricsController @Inject() (
+    plugin:        MetricsPlugin,
+    configuration: Configuration,
+    cc:            ControllerComponents
+)
+  extends AbstractController(cc) with PrometheusSupport {
+  private def defaultFormat: String = configuration.getAndValidate[String]("org.zalando.markscheider.defaultFormat", Set("zmon", "prometheus"))
+
+  def serializePrometheus(registry: MetricRegistry): Result = {
```
Comment (matterche) [inline app/org/zalando/markscheider/MetricsController.scala]: This one could be private, right?
```suggestion
  private def serializePrometheus(registry: MetricRegistry): Result = {
```

File: app/org/zalando/markscheider/MetricsController.scala [hunk 2]
```diff
@@ -2,26 +2,70 @@ package org.zalando.markscheider
...
+    }).flatten
+
+    val stringWriter = new StringWriter()
+    for { entry <- histograms ++ timers } {
+      stringWriter.write(entry.toString)
+    }
+    Ok(stringWriter.toString).withHeaders(HeaderNames.CACHE_CONTROL -> "must-revalidate,no-cache,no-store").as("text/plain; version=0.0.4")
+  }
+
+  def serializeJson(mapper: ObjectMapper): Result = {
```
Comment (matterche) [inline app/org/zalando/markscheider/MetricsController.scala]: This one could be private, right?

```suggestion
  private def serializeJson(mapper: ObjectMapper): Result = {
```

File: test/org/zalando/markscheider/PrometheusSupportSpec.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package org.zalando.markscheider
+
+import com.codahale.metrics.{ ExponentiallyDecayingReservoir, Histogram, Reservoir }
```
Comment (matterche) [inline test/org/zalando/markscheider/PrometheusSupportSpec.scala]: Unused import
```suggestion
import com.codahale.metrics.{ ExponentiallyDecayingReservoir, Histogram }
```

File: README.md [hunk 4]
```diff
@@ -17,34 +17,27 @@ Add metrics-play dependency in your `build.sbt`:
 
 ```scala
 libraryDependencies += Seq(
-  "org.zalando" %% "markscheider" % "2.5.2"
+  "org.zalando" %% "markscheider" % "2.7.0"
```
Comment (matterche) [inline README.md]: 2.7.0 is already released. I just forgot to open a PR ☺️

```suggestion
  "org.zalando" %% "markscheider" % "2.7.1"
```

Please could you also adjust the version.sbt?

**Review by matterche — CHANGES_REQUESTED** [review]
Just some minor comments

File: app/org/zalando/markscheider/MetricsController.scala (hunk 1 above)
Comment (lenalebt) [reply app/org/zalando/markscheider/MetricsController.scala]: absolutely.

File: app/org/zalando/markscheider/MetricsController.scala (hunk 2 above)
Comment (lenalebt) [reply app/org/zalando/markscheider/MetricsController.scala]: yep

File: test/org/zalando/markscheider/PrometheusSupportSpec.scala (hunk 3 above)
Comment (lenalebt) [reply test/org/zalando/markscheider/PrometheusSupportSpec.scala]: deleted

File: README.md (hunk 4 above)
Comment (lenalebt) [reply README.md]: np, updated to 2.7.1 - the version in version.sbt, too.

**Review by matterche — APPROVED** [review]

**Comment by matterche** [comment]
@musiKk would you approving the PR? There's currently no active second maintainer from Zalando.

**Comment by matterche** [comment]
👍

**Comment by musiKk** [comment]
👍

**Comment by lenalebt** [comment]
nice, thanks :)

**Comment by matterche** [comment]
Thanks for the contribution @lenalebt Artifacts should be available now.

Btw. I had to revert the update of sbt-pgp because it broke the release process. See #28

**Comment by lenalebt** [comment]
Oh, sorry. I used that one in another project already with no problems, but your mileage may vary.

**Pull request merged**